model_specific/SM/decays/standard_model_decays.hpp

```diff
@@ -0,0 +1,221 @@
+
+template<>
+Decay_amplitude_SFF Standard_model_decays::calculate_amplitude<standard_model_cxx_diagrams::fields::hh, typename standard_model_cxx_diagrams::fields::bar<standard_model_cxx_diagrams::fields::Fv>::type, standard_model_cxx_diagrams::fields::Fv>(const standard_model_cxx_diagrams::context_base&, const typename cxx_diagrams::field_indices<standard_model_cxx_diagrams::fields::hh >::type&, const typename cxx_diagrams::field_indices<typename standard_model_cxx_diagrams::fields::bar<standard_model_cxx_diagrams::fields::Fv>::type >::type&, const typename cxx_diagrams::field_indices<standard_model_cxx_diagrams::fields::Fv >::type&) const;
+
+template<>
+Decay_amplitude_SFF Standard_model_decays::calculate_amplitude<standard_model_cxx_diagrams::fields::hh, typename standard_model_cxx_diagrams::fields::bar<standard_model_cxx_diagrams::fields::Fd>::type, standard_model_cxx_diagrams::fields::Fd>(const standard_model_cxx_diagrams::context_base&, const typename cxx_diagrams::field_indices<standard_model_cxx_diagrams::fields::hh >::type&, const typename cxx_diagrams::field_indices<typename standard_model_cxx_diagrams::fields::bar<standard_model_cxx_diagrams::fields::Fd>::type >::type&, const typename cxx_diagrams::field_indices<standard_model_cxx_diagrams::fields::Fd >::type&) const;
+
+template<>
+Decay_amplitude_SFF Standard_model_decays::calculate_amplitude<standard_model_cxx_diagrams::fields::hh, typename standard_model_cxx_diagrams::fields::bar<standard_model_cxx_diagrams::fields::Fu>::type, standard_model_cxx_diagrams::fields::Fu>(const standard_model_cxx_diagrams::context_base&, const typename cxx_diagrams::field_indices<standard_model_cxx_diagrams::fields::hh >::type&, const typename cxx_diagrams::field_indices<typename standard_model_cxx_diagrams::fields::bar<standard_model_cxx_diagrams::fields::Fu>::type >::type&, const typename cxx_diagrams::field_indices<standard_model_cxx_diagrams::fields::Fu >::type&) const;
+
+
+template<typename FieldIn, typename FieldOut1, typename FieldOut2>
+double
+Standard_model_decays::amplitude_squared(standard_model_cxx_diagrams::context_base const& context,
+                  typename cxx_diagrams::field_indices<FieldIn>::type const& indexIn,
+                  typename cxx_diagrams::field_indices<FieldOut1>::type const& indexOut1,
+                  typename cxx_diagrams::field_indices<FieldOut2>::type const& indexOut2) const
+{
+
+   const auto mat_elem = calculate_amplitude<FieldIn, FieldOut1, FieldOut2>(
+      context, indexIn, indexOut1, indexOut2);
+   return mat_elem.square();
+}
+
+// generic decay of FieldIn -> FieldOut1 FieldOut2
+template<typename FieldIn, typename FieldOut1, typename FieldOut2>
+double Standard_model_decays::get_partial_width(
```

Comment:
I think thif function can be `const`.

Reply:
This is unfortunate. The naming comes from Dylan. But now `get_partial_width` also sets the effective coupling block (member variable `effhiggscouplings_block_input`). So it's no longer const.

Reply:
Sorry, I don't see where this member is modified. I have the feeling that github does not point to the right place in the "conversation". When I go to "changed files" the place where I'm referring to is the `get_partial_width` template. It does not seem to modify any members. Am I missing something?

Reply:
Ok, so due to the modification the function cannot be made `const`, I see.